sys/userlib/Cargo.toml

```diff
@@ -6,6 +6,7 @@ edition = "2021"
 [features]
 default = ["critical-section"]
 panic-messages = []
+no-panic = []
```

Comment:
Take it or leave it: it might be nice if the crate's top-level documentation mentioned both of these feature flags and their interactions. 

In particular, I notice that `no-panic` will take priority over `panic-messages`. I think this is probably a fine behavior, but it probably ought to be written down?

Alternatively, we could do something like this:
```rust
#[cfg(all(feature = "panic-messages", feature = "no-panic"))]
compile_error!(
    "both the `userlib/panic-messages` and `userlib/no-panic` feature flags \
    are enabled! this doesn't make a lot of sense and is probably not what \
    you wanted."
);
```

IMO, enabling both these features is _probably_ a mistake, and I might prefer to know that I made the mistake more than I'd want it to still build. On the other hand, because of cross-crate feature unification, it does seem possible to have a situation where both features are enabled for a task, because a dependency enables one of them. But, a library dependency generally shouldn't do that...I dunno.

This might be more relevant if we consider a mechanism for Idol-generated APIs to be marked as non-panicking, so that we can avoid generating retry code for non-idempotentent IPCs served by  tasks that are not able to panic?

Reply:
I like the build script suggestion to force them to be exclusive for now, until we find a reason for them not to be.